de4a-connector/src/main/java/eu/de4a/connector/as4/client/DomibusGatewayClient.java

```diff
@@ -51,179 +56,180 @@
+                domibusRequestRepository.save(req);                    
+                processMessage(id);
+            });
+        }
+    }
+
+    @Override
+    public void sendMessage(String sender, NodeInfo nodeInfo, Element requestUsuario,
+            List<TCPayload> payloads, boolean isRequest) throws MEOutgoingException {
+        String idMessageAttached = "cid:message";
+        String idCanonical = "cid:"
+                + (isRequest ? DE4AConstants.TAG_EVIDENCE_REQUEST : DE4AConstants.TAG_EVIDENCE_RESPONSE);
+
+        Messaging messageHeader = MessageFactory.makeMessage(sender, nodeInfo.getParticipantIdentifier(),
+                nodeInfo.getDocumentIdentifier(), nodeInfo.getProcessIdentifier(),
+                getAttachments(idMessageAttached, idCanonical));
+
+        List<LargePayloadType> bodies = new ArrayList<>();
+        LargePayloadType payload = new LargePayloadType();
+        payload.setContentType(MediaType.APPLICATION_XML_VALUE);
+        payload.setPayloadId(idMessageAttached);
+        DataSource source = null;
+        try {
+            source = new ByteArrayDataSource(DOMUtils.documentToByte(requestUsuario.getOwnerDocument()),
+                    MediaType.APPLICATION_XML_VALUE);
+        } catch (MessageException e) {
+            throw new MEOutgoingException(e.getMessage(), e);
+        }
+        payload.setValue(new DataHandler(source));
+        bodies.add(payload);
+        if (payloads != null) {
+            payloads.forEach(p -> {
+                LargePayloadType payloadtmp = new LargePayloadType();
+                payloadtmp.setContentType(p.getMimeType());
+                String cid = p.getContentID().startsWith("cid:") ? p.getContentID() : "cid:" + p.getContentID();
+                payloadtmp.setPayloadId(cid);
+                DataSource sourcetmp = new ByteArrayDataSource(p.getValue(), p.getMimeType());
+                payloadtmp.setValue(new DataHandler(sourcetmp));
+                bodies.add(payloadtmp);
+            });
+        }
+        try {
+            domibusClientWS.submitMessage(messageHeader, bodies, domibusEndpoint);
+        } catch (DomibusException e) {
+            throw new MEOutgoingException(e.getMessage(), e);
+        }
+    }
+
+    public ResponseWrapper processResponseAs4(IncomingEDMResponse responseas4) {
+        LOGGER.debug("Processing AS4 response...");
+        
+        ConnectorException ex = new ConnectorException().withLayer(LayerError.INTERNAL_FAILURE)
+            .withFamily(FamilyErrorType.CONVERSION_ERROR)
+            .withModule(ExternalModuleError.CONNECTOR_DR)
+            .withHttpStatus(HttpStatus.OK);
+        
+        ResponseWrapper responsewrapper = new ResponseWrapper();
+        responseas4.getAllAttachments().forEachValue(a -> {
+            try {
+                responsewrapper.addAttached(
+                        FileUtils.getMultipart(a.getContentID().replace("cid:", ""), a.getMimeTypeString(), a.getData().bytes()));
```

Comment:
I suggest you remove it only, if the value "startsWith" "cid:" to avoid removing some "cid:" in the middle

Reply:
Good point, but right now we have the tags naming under control, that case wouldn´t be possible